### PR TITLE
Introduces some basic arg-parsing with getopts.

### PR DESCRIPTION
**Pull request by jloehel**
Signed-off-by: Jürgen Löhel <jloehel@suse.com>

### CODE REVIEW

File: stsx [hunk 1]
```diff
@@ -28,67 +28,114 @@
...
+  xs3simdev)      simulation="devices" delay=5 ;;
+  xs3simplatform) simulation="platform" delay=5 ;;
+  xs3simcore)     simulation="core" delay=5; ;;
+  xs4simcpu)      simulation="processors" delay=0 ;;
+  x)              error "Argument missing. Please specify a method.";;
+  *)              error "Unknown method $method"
 esac
 
+
+if ! [ "$watchdog_port" -eq "$watchdog_port" ] 2>/dev/null;then
```
Comment (sndirsch) [inline stsx]: Don't understand this line. This will always evaluate to false, right? What's the purpose behind this if-clause?

File: stsx [hunk 2]
```diff
@@ -28,67 +28,114 @@
...
+  error "Invalid port number"
+fi
+
+if ! [ "$watchdog_port" -ge 1 -a "$watchdog_port" -le 65535 ]; then
```
Comment (sndirsch) [inline stsx]: Don't understand this line. This will always evaluate to false, right? What's the purpose behind this if-clause?

File: stsx (hunk 1 above)
Comment (sndirsch) [inline stsx]: I don't understand this line.

**Review by sndirsch — COMMENTED** [review]
Looks good. I just don't understand this one if-clause.

File: stsx (hunk 1 above)
Comment (tiwai) [reply stsx]: It's for checking whether the given argument is an integer. 

However, don't forget that nc can accept a string service as port as well, not only the port integer number.

Comment (sndirsch) [reply stsx]: Ah! Understood! But then in that case I suggest to simply remove the check.

Comment (jloehel) [reply stsx]: Thank you Takashi. I didn't know that this ```nc google.de http``` is possible. I will remove it.

**Pull request merged**